core/trino-main/src/main/java/io/trino/operator/TableDeleteOperator.java

```diff
@@ -13,37 +13,29 @@
  */
 package io.trino.operator;
 
-import com.google.common.collect.ImmutableList;
 import io.trino.Session;
 import io.trino.metadata.Metadata;
 import io.trino.metadata.TableHandle;
 import io.trino.spi.Page;
-import io.trino.spi.PageBuilder;
-import io.trino.spi.block.BlockBuilder;
-import io.trino.spi.type.Type;
 import io.trino.sql.planner.plan.PlanNodeId;
 
-import java.util.List;
-import java.util.OptionalLong;
-
 import static com.google.common.base.Preconditions.checkState;
-import static io.trino.spi.type.BigintType.BIGINT;
 import static java.util.Objects.requireNonNull;
 
 public class TableDeleteOperator
         implements Operator
 {
-    public static final List<Type> TYPES = ImmutableList.of(BIGINT);
+    private final TableMetadataOperator metadataOperator;
 
     public static class TableDeleteOperatorFactory
             implements OperatorFactory
     {
-        private final int operatorId;
-        private final PlanNodeId planNodeId;
-        private final Metadata metadata;
-        private final Session session;
-        private final TableHandle tableHandle;
-        private boolean closed;
+        protected final int operatorId;
+        protected final PlanNodeId planNodeId;
+        protected final Metadata metadata;
+        protected final Session session;
+        protected final TableHandle tableHandle;
+        protected volatile boolean closed;
```

Comment:
Why `protected`?

Reply:
It's leftovers after refactoring.
Fixed.